src/tools/erthost/erthost.cpp

```diff
@@ -141,6 +146,85 @@ static int run(const char* path, bool simulate)
     return return_value;
 }
 
+static void _trim_space(string& str)
+{
+    int (&isspace)(int) = ::isspace; // isspace is overloaded, select the wanted
+    str.erase(find_if_not(str.rbegin(), str.rend(), isspace).base(), str.end());
+    str.erase(str.begin(), find_if_not(str.begin(), str.end(), isspace));
+}
+
+static void _trim_prefix(string& str, string_view prefix)
+{
+    if (str.compare(0, prefix.size(), prefix) == 0)
+        str.erase(0, prefix.size());
+}
+
+static void _log(
+    void* /*context*/,
+    bool is_enclave,
+    const tm* /*t*/,
+    long /*usecs*/,
+    oe_log_level_t level,
+    uint64_t /*host_thread_id*/,
+    const char* message)
+{
+    assert(message && *message);
+    if (level > oe_get_current_logging_level())
+        return;
+    const auto level_string = oe_log_level_strings[level];
+
+    // split message of the form "log message ... [/path/to/source:func:line]"
+    static const regex re_message(R"(([^]+) \[(.+):(\w+:\d+)]\n)");
+    cmatch ma_message;
+    if (!regex_match(message, ma_message, re_message))
+    {
+        // not this form, so just print it
+        cout << level_string << ": " << message << '\n';
+        return;
+    }
+    string msg = ma_message[1];
+    string path = ma_message[2];
+    const auto& func_and_line = ma_message[3];
+
+    // strip enclave name
+    if (is_enclave)
+        msg.erase(0, msg.find(':') + 1);
+
+    // Check if the message contains the same OE error value as the last one.
+    // This is a heuristic, but should be good enough.
```

Comment:
This seems a bit sophisticated to change the log output, sure it's still easier / less invasive than just patching OE's log code? Or can't we reduce this code a bit and use `OE_LOG_FORMAT` to influence the output style?

Reply:
Considered both. Patching OE requires touching multiple files and still requires at least some of the parsing. OE_LOG_FORMAT is great for reformatting, but poor for omitting infos.